src/mongo/delegates/Views.class.php

```diff
@@ -396,58 +406,16 @@ public function generateView($viewId,$resource=null,$context=null,$queueName=nul
             $t = new \Tripod\Timer();
             $t->start();
 
-            $from = $this->getFromCollectionForViewSpec($viewSpec);
+            $from = $viewSpec['from'];
             $collection = $this->config->getCollectionForView($this->storeName, $viewId);
 
+            $this->ensureIndices($viewSpec, $collection);
```

Comment:
This seems to be happening on every update.  Is that necessary?  Once configured, the indices should take care of themselves...